Tests/MongoSwiftTests/ReadWriteConcernTests.swift

```diff
@@ -79,11 +125,53 @@ final class ReadWriteConcernTests: XCTestCase {
         client3.close()
     }
 
+    func testClientWriteConcern() throws {
+        // create a client with no options and check its RC
+        let client1 = try MongoClient()
```

Comment:
I know we've spoken about this before, but `close` is called in the clients deinitializer so the calls to that are somewhat superfluous. Also, visually if you broke these up into sub-scopes, it might be easier to scan which test is which:
```
  { 
       // create a client with w: 1 and check its WC
       let client2 = try MongoClient(options: ClientOptions(writeConcern: WriteConcern(w: 1)))
       // although w:1 is default, if it is explicitly provided it should be set
       expect(client2.writeConcern?.w).to(equal(1))

       // expect that a DB created from this client inherits its WC
       let db3 = try client2.db("test")
       expect(db3.writeConcern?.w).to(equal(1))

       // expect that a DB created from this client can override the client's WC
       let db4 = try client2.db("test", options: DatabaseOptions(writeConcern: WriteConcern(w: 2)))
       expect(db4.writeConcern?.w).to(equal(2))
  }
```